### ISSUE TITLE
a question about prepossessing

### ISSUE THREAD

**Issue by hanquansanren** [fh2019ustc/DocGeoNet#2]
hello hao,
Thanks for your third awesome work for document image dewarping. I have a simple question, that is,
In your section 5.2 of the paper , you pointed out that `The preprocessing module and the following rectification module are trained independently`. as shown in the following:

![image](https://user-images.githubusercontent.com/36793626/197205087-0a3b2d8d-772e-4b24-a8e4-94cb782b1688.png)
My question is why don't you train the whole flow jointly? Have you made any trials to verify the 2-stage rectification would be better?

**Comment by hanquansanren** [comment]
sorry, the issue title should be preprocessing, instead of prepossessing

**Comment by fh2019ustc** [comment]
Hi, thanks for your attention to our work, and this is a nice concern.
In fact, the whole network can be trained jointly.
We perform the 2-stage rectification here based on the following reasons， 
1. The gradients can be backpropagated to the prepossessing network due to the nondifferentiable mask. Because the distorted image should be multiplied by the document mask（0 denotes the background and 1 denotes the foreground document region）.
2. Joint training would consume more GPUs.
I hope this can help you.

**Comment by hanquansanren** [comment]
I see, it helps me a lot, thanks for your kind response
